test/UnitTest.cs

```diff
@@ -119,6 +119,17 @@ public static void CountOccurrences(string text, string pattern, int expected)
             var count = CountStr(text, pattern);
             Assert.AreEqual(expected, count);
         }
+        public static bool OccurenceWithOrder(string text, List<string> patternlist)
```

Comment:
we shall assert that the order is underneath join? try to check the plan (not the plan string) directly. - you can extend the ExecSQL() interface if we can't get the plan.